test/boost/network_topology_strategy_test.cc

```diff
@@ -480,6 +530,148 @@ SEASTAR_THREAD_TEST_CASE(NetworkTopologyStrategy_tablets_test) {
     }
 }
 
+SEASTAR_THREAD_TEST_CASE(NetworkTopologyStrategy_tablet_allocation_balancing_test) {
+    auto my_address = gms::inet_address("localhost");
+
+    // Create the RackInferringSnitch
+    snitch_config cfg;
+    cfg.listen_address = my_address;
+    cfg.broadcast_address = my_address;
+    cfg.name = "RackInferringSnitch";
+    sharded<snitch_ptr> snitch;
+    snitch.start(cfg).get();
+    auto stop_snitch = defer([&snitch] { snitch.stop().get(); });
+    snitch.invoke_on_all(&snitch_ptr::start).get();
+
+    locator::token_metadata::config tm_cfg;
+    tm_cfg.topo_cfg.this_endpoint = my_address;
+    tm_cfg.topo_cfg.local_dc_rack = { snitch.local()->get_datacenter(), snitch.local()->get_rack() };
+
+    // Generate a random cluster
```

Comment:
Even in debug mode this test takes 0.5 seconds to run. You use 4 random integers here:

- num_racks -- 4 values
- nodes_per_rack -- 4 values
- shard_count -- 4 values
- rf_per_dc (below) -- 13 values

Even if performing exhaustive test of _all_ values you get 4 * 4 * 4 * 13 * 0.5sec = ~6minutes. Pretty fast :)
It's worth either turning this test into exhaustive one, or at least checking ~100 "random" iterations in a loop

Reply:
with aarch64/debug it will take a year :)
I'm not sure we need need to exhaust all cases every time the test runs, as we'll get coverage over time by running this test on many occasions.

But running multiple iterations with random choices to increase coverage isn't a bad idea.

Reply:
in next version I'll refactor the tablet balancing test and call the inner function 100 times to test 100 random cases every run. That said, it makes it harder to reproduce an issue if it is generated on the Nth iteration, based on the random seed, so I'll add a printout for each call's seed so it can be reproduced this way.

Reply:
done in v4